s3/libirods_s3.cpp

```diff
@@ -431,106 +433,125 @@ irods::error parseS3Path (
+    if (!fptr) {
+        return ERROR(SYS_CONFIG_FILE_ERR, fmt::format(
+                     "[resource_name={}] Failed to open S3 auth file: \"{}\", errno = \"{}\".",
+                     resource_name, _filename, strerror(errno)));
+    }
 
-        while ((lineLen = getLine (fptr, inbuf, MAX_NAME_LEN)) > 0) {
-            char *inPtr = inbuf;
-            if (linecnt == 0) {
-                while ((bytesCopied = getStrInBuf (&inPtr, access_key_id, &lineLen, S3_MAX_KEY_SIZE)) > 0) {
-                    linecnt ++;
-                    break;
-                }
-            } else if (linecnt == 1) {
-                while ((bytesCopied = getStrInBuf (&inPtr, secret_access_key, &lineLen, S3_MAX_KEY_SIZE)) > 0) {
-                    linecnt ++;
-                    break;
-                }
+    while ((lineLen = getLine (fptr, inbuf, MAX_NAME_LEN)) > 0) {
+        char *inPtr = inbuf;
+        if (linecnt == 0) {
+            while ((bytesCopied = getStrInBuf (&inPtr, access_key_id, &lineLen, S3_MAX_KEY_SIZE)) > 0) {
+                linecnt ++;
+                break;
+            }
+        } else if (linecnt == 1) {
+            while ((bytesCopied = getStrInBuf (&inPtr, secret_access_key, &lineLen, S3_MAX_KEY_SIZE)) > 0) {
+                linecnt ++;
+                break;
             }
         }
-        if ((result = ASSERT_ERROR(linecnt == 2, SYS_CONFIG_FILE_ERR, "[resource_name=%s] Read %d lines in the auth file. Expected 2.", resource_name.c_str(),
-                                   linecnt)).ok())  {
-            _rtn_key_id = access_key_id;
-            _rtn_access_key = secret_access_key;
-        }
-        return result;
     }
 
-    if (fptr) {
-        fclose(fptr);
+    if (linecnt != 2) {
+        return ERROR(SYS_CONFIG_FILE_ERR, fmt::format(
+                     "[resource_name={}] Read {} lines in the auth file. Expected 2.",
+                     resource_name, linecnt));
     }
 
-    return result;
-}
+    _rtn_key_id = access_key_id;
+    _rtn_access_key = secret_access_key;
+
+    return SUCCESS();
+} // readS3AuthInfo
 
 /// @brief Retrieves the auth info from either the environment or the resource's specified auth file and set the appropriate
 /// fields in the property map
 irods::error s3ReadAuthInfo(
     irods::plugin_property_map& _prop_map)
 {
-    irods::error result = SUCCESS();
-    irods::error ret;
     char* tmpPtr;
     std::string key_id;
     std::string access_key;
 
     std::string resource_name = get_resource_name(_prop_map);
-
+    // TODO: why do we not return immediately? key_id and acccess_key are not referenced
+    // again, and there is no failure in the event that either key is missing from the
+    // environment.
     if ((tmpPtr = getenv(s3_key_id.c_str())) != NULL) {
         key_id = tmpPtr;
         if ((tmpPtr = getenv(s3_access_key.c_str())) != NULL) {
             access_key = tmpPtr;
         }
```

Comment:
Agreed.  I am not sure what the point of this is.  Why are environment variables being used?  And why set key_id and access_key if they are not going to be used?

If I understand this correctly, if the environment variables are set, they take precedence which would mean that you could not have multiple key pairs for multiple resources.

I would say torch this part.

Reply:
I guess the question is whether or not we plan to support environment variables as an alternative to the keypair file. As it stands, the environment variable use case is not being fully supported, but it looks like it may have been in the past.

I'm okay with torching it but just know that we will be officially dropping that use case instead of having a broken feature (in which case I will file an issue). Thoughts?

Reply:
After discussing, we have decided that we want to drop this. I will do so in the next commit